java-frontend/src/main/java/org/sonar/java/se/ExplodedGraphWalker.java

```diff
@@ -293,6 +302,15 @@ private void handleEndOfExecutionPath(boolean interrupted) {
     setNode(savedNode);
   }
 
+  public void addExceptionalYield(SymbolicValue target, ProgramState exceptionalState, String exceptionFullyQualifiedName, SECheck check) {
+    if (methodBehavior != null && methodBehavior.parameters().contains(target)) {
+      Type exceptionType = semanticModel.getClassType(exceptionFullyQualifiedName);
+      ProgramState newExceptionalState = exceptionalState.clearStack().stackValue(constraintManager.createExceptionalSymbolicValue(exceptionType));
+      ExplodedGraph.Node exitNode = explodedGraph.node(new ProgramPoint(exitBlock), newExceptionalState);
```

Comment:
as discussed : the exit block here may not be required (but if it is : it should probably be tested)